cheroot/test/test_server.py

```diff
@@ -439,3 +441,88 @@ def many_open_sockets(request, resource_limit):
         # Close our open resources
         for test_socket in test_sockets:
             test_socket.close()
+
+
+@pytest.mark.parametrize(
+    ('minthreads', 'maxthreads'),
+    (
+        (1, -2),  # the docstring only mentions -1 to mean "no max", but other
+                  # negative numbers should also work
+        (1, -1),
+        (1, 1),
+        (1, 2),
+        (1, float('inf')),
+        (2, -2),
+        (2, -1),
+        (2, 2),
+        (2, float('inf')),
+    ),
+)
+def test_threadpool_threadrange_set(minthreads, maxthreads):
+    """Test setting the number of threads in a ThreadPool.
+
+    The ThreadPool should properly set the min+max number of the threads to use
+    in the pool if those limits are valid.
+    """
+    tp = ThreadPool(
+        server=None,
+        min=minthreads,
+        max=maxthreads,
+    )
+    assert tp.min == minthreads
+    if maxthreads < 0:
+        assert tp.max == float('inf')
+    else:
+        assert tp.max == maxthreads
```

Comment:
It's best to have a clear expectation regarding what the max is, instead of putting logic into the test.
```suggestion
@pytest.mark.parametrize(
    ('minthreads', 'maxthreads', 'inited_maxthreads'),
    (
        (
            # NOTE: The docstring only mentions -1 to mean "no max", but other
            # NOTE: negative numbers should also work.
            1,
            -2,
            float('inf'),
        ),
        (1, -1, float('inf')),
        (1, 1, 1),
        (1, 2, 2),
        (1, float('inf'), float('inf')),
        (2, -2, float('inf')),
        (2, -1, float('inf')),
        (2, 2, 2),
        (2, float('inf'), float('inf')),
    ),
)
def test_threadpool_threadrange_set(minthreads, maxthreads, inited_maxthreads):
    """Test setting the number of threads in a ThreadPool.

    The ThreadPool should properly set the min+max number of the threads to use
    in the pool if those limits are valid.
    """
    tp = ThreadPool(
        server=None,
        min=minthreads,
        max=maxthreads,
    )
    assert tp.min == minthreads
    assert tp.max == inited_maxthreads
```